doc/cprover-manual/restrict-function-pointer.md

```diff
@@ -0,0 +1,180 @@
+[CPROVER Manual TOC](../../)
+
+## Restricting function pointers
+
+In this document, we describe the `goto-instrument` feature to replace calls
+through function pointers by case distinctions over calls to given sets of
+functions.
+
+### Motivation
+
+The CPROVER framework includes a goto program transformation pass
+`remove_function_pointers()` to resolve calls to function pointers to direct
+function calls. The pass is needed by `cbmc`, as symbolic execution itself can't
+handle calls to function pointers. In practice, the transformation pass works as
+follows:
```

Comment:
:bulb: this is kind of an impl detail - and as such I'd be tempted to bury it at the _bottom_ of the docs

Reply:
Will leave updating the documentation to the next PR which changes some things.